processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -304,31 +306,79 @@ private void handleDefinedNestedTargetMapping(Set<String> handledTargets) {
 
                 Map<Parameter, MappingOptions> optionsBySourceParam = entryByTP.getValue().groupBySourceParameter();
                 boolean forceUpdateMethod = optionsBySourceParam.keySet().size() > 1;
+                PropertyEntry targetProperty = entryByTP.getKey();
                 for ( Entry<Parameter, MappingOptions> entryByParam : optionsBySourceParam.entrySet() ) {
```

Comment:
I  think you did a great job already 👍 .. There is room for improvement. Need to take some time to check the generated code. In the mean time I pushed a small nitty-witty check style / javadoc commit..